src/cluster/sharding.rst

```diff
@@ -179,123 +328,114 @@ looks like this:
+    $ curl -X PUT $COUCH_URL:5986/_nodes/{name}@{address} \
+        -d '{ \
+            "_id": "{name}@{address}",
+            "_rev": "{rev}",
+            "zone": "{zone-name}"
+            }'
 
-.. _cluster/sharding/views:
+In the config file (local.ini or default.ini) of each node, define a
+consistent cluster-wide setting like:
 
-Views
-=====
+::
 
-The views need to be moved together with the shards. If you do not, then
-CouchDB will rebuild them and this will take time if you have a lot of
-documents.
+    [cluster]
+    placement = {zone-name-1}:2,{zone-name-2}:1
 
-The views are stored in ``data/.shards``.
+In this example, it will ensure that two replicas for a shard will be
+hosted on nodes with the zone attribute set to ``{zone-name-1}`` and one
+replica will be hosted on a new with the zone attribute set to
+``{zone-name-2}``.
 
-It is possible to not move the views and let CouchDB rebuild the view every
-time you move a shard. As this can take quite some time, it is not recommended.
+Note that you can also use this system to ensure certain nodes in the
+cluster do not host any replicas for newly created databases, by giving
+them a zone attribute that does not appear in the ``[cluster]``
+placement string.
 
-.. _cluster/sharding/preshard:
+You can also specify zones on a per-database basis by specifying the
+zone as a query parameter when the database is created:
 
-Reshard? No, Preshard!
-======================
+.. code:: bash
 
-Reshard? Nope. It cannot be done. So do not create databases with too few
-shards.
+    curl -X PUT $COUCH_URL:5984/{dbName}?zone={zone}
 
-If you can not scale out more because you set the number of shards too low, then
-you need to create a new cluster and migrate over.
+Resharding
+----------
 
-#. Build a cluster with enough nodes to handle one copy of your data.
-#. Create a database with the same name, n=1 and with enough shards so you do
-   not have to do this again.
-#. Set up 2 way replication between the 2 clusters.
-#. Let it sync.
-#. Tell clients to use both the clusters.
-#. Add some nodes to the new cluster and add them as replicas.
-#. Remove some nodes from the old cluster.
-#. Repeat 6 and 7 until you have enough nodes in the new cluster to have 3
-   replicas of every shard.
-#. Redirect all clients to the new cluster
-#. Turn off the 2 way replication between the clusters.
-#. Shut down the old cluster and add the servers as new nodes to the new
-   cluster.
-#. Relax!
+Shard settings for databases can only be set when the database is
+created, precluding live resharding. Instead, to reshard a database, it
+must be regenerated. Here are the steps:
 
-Creating more shards than you need and then move the shards around is called
-presharding. The number of shards you need depends on how much data you are
-going to store. But, creating too many shards increases the complexity without
-any real gain. You might even get lower performance. As an example of this, we
```

Comment:
We probably don't want to lose this idea entirely from the docs. I recall from my time at Cloudant creating DBs with very high `q` values under the assumption that new HW was coming soon, and the shards would be re-balanced to the new machines once they were available. I actually forgot we did this until I read this section.

Reply:
On second thought, "presharding" is more related to the concept of CouchDB's sharding in general - we pre-shard databases, meaning we don't re-shard a database after its creation, we decide at creation time where shards go.

Anyway, let's not get into specific recommendations without community consensus. Starting a thread on user@ or dev@ asking for sharding advice based on DB size, doc size, etc. would probably be a good first step.